### ISSUE TITLE
W3 Validation Fails

### ISSUE THREAD

**Issue by gatzkerob** [simple-icons/simple-icons#503]
`aria-labelledby="title"` is failing w3 validation:

"Error: The aria-labelledby attribute must point to an element in the same document"

You could add an `id="title"` to `<title>,` but if you use multiple icons on the same page you'll end up with:

"Error: Duplicate ID title"

I don't think there's a way get around this as of this post, aside from using ARIA 1.0 while omitting the `aria-labelledby` attribute.

Something to keep in mind.

More info here: [https://www.w3.org/TR/svg-aam-1.0/](https://www.w3.org/TR/svg-aam-1.0/)

And here: [https://simplyaccessible.com/article/7-solutions-svgs/](https://simplyaccessible.com/article/7-solutions-svgs/)

EDIT: Ok this is embarrassing. Yeah, a unique id for each title would work.

**Comment by ericcornelissen** [comment]
Perhaps we can use an id that is different for each icon? Something like `simpleicon-[ICON NAME]`, perhaps with a more complicated prefix to avoid duplicate ids on the pages of people using the icons...

**Comment by birjolaxew** [comment]
@gatzkerob Thanks for the heads up!

Ideally screen readers would be "clever" enough to simple understand that the `<title>` describes the SVG. Unfortunately, [they aren't](https://www.paciellogroup.com/blog/2013/12/using-aria-enhance-svg-accessibility/).

I like @ericcornelissen's suggestion. A unique-per-icon ID should be sufficient to avoid borking everything too much.

**Comment by birjolaxew** [comment]
It should be somewhat trivial to write a script that adds the ID to the existing icons. I'll have a look at it later.

**Comment by birjolaxew** [comment]
Leaving #504 open for comments for a day. If you could test it @gatzkerob that'd be much appreciated :)

**Comment by ericcornelissen** [comment]
> I think W3 will still fail it in case you include the same icon multiple times, but I don't see a way to fix that.

It might be an idea to generate a random id upon downloading the `.svg` file? Although I think for now #504 is a good enough solution 👍

**Comment by birjolaxew** [comment]
The changes should now be published on the live site. If you could test it @gatzkerob that'd be much appreciated
